### DISCUSSION TITLE
Struggling with textual console for debugging

### DISCUSSION THREAD

**Discussion by jspv** [Textualize/textual#1063] [Q&A] [answered]
My app is pretty basic and I write my code similar to:

```
class MyApp(app):
  # cool textual stuff

def main():
    # do lots of non-textual stuff - parse arguments with argparse, verify dependencies, find ducks, line them up, etc.

   app = MyApp()
   app.run()
```

My tools are formatted as a package, this allows me do do things like relative imports, and I often run the tools via:

```
from tool.tool import main

if __name__ == '__main__':
    main()
```

I'm having a lot of trouble debugging getting apps to new textual console for debugging.  

After much struggling with `textual run tool`, `textual run tool/tool.py` (can't do this as it breaks relative imports), etc. I was finally able get `textual run tool.tool:MyApp` to run, but that of course cuts out all of the code in main(); I could move all that code into the `__init__` for MyApp (which I did) - but then the textual command gets all confused when I try and pass arguments to my tool.  I also feel like I'm doing something wrong reformatting my code so much just to get textual debugging working.  Does this new version of textual support any logfile options similar to the old 0.1 branch?

I'd really appreciate if someone please can show an example of using `textual run --dev` with a script that takes arguments or point me in a different direction to get the textual debug log.  

Many thanks.

**Comment by lllama** [comment, marked as answer]
Given you're doing some things in `main` you might want to try setting the `TEXTUAL` environment variable. It looks like if you set it to `debug,devtools` then you'll get the same effect as running with `--dev`.
